Extensions/TweenBehavior/tweenruntimebehavior.js

```diff
@@ -0,0 +1,301 @@
+      return true
+    }
+  }
+  return false;
+};
+
+gdjs.TweenRuntimeBehavior._tweenIsPlaying = (owner, identifier) => {
+  return gdjs.TweenRuntimeBehavior._tweenList[owner][identifier].instance.isPlaying();
+};
+
+gdjs.TweenRuntimeBehavior._pauseTween = (owner, identifier) => {
+  return gdjs.TweenRuntimeBehavior._tweenList[owner][identifier].instance.pause();
+};
+
+gdjs.TweenRuntimeBehavior._resumeTween = (owner, identifier) => {
+  return gdjs.TweenRuntimeBehavior._tweenList[owner][identifier].instance.resume();
+};
+
+gdjs.TweenRuntimeBehavior._stopTween = (owner, identifier, JumpToDest) => {
+  return gdjs.TweenRuntimeBehavior._tweenList[owner][identifier].instance.stop(JumpToDest);
+};
+
+gdjs.TweenRuntimeBehavior._setTweenFinished = (owner, identifier, hasFinished) => {
+  gdjs.TweenRuntimeBehavior._tweenList[owner][identifier].hasFinished = hasFinished;
+};
+
+gdjs.TweenRuntimeBehavior._tweenHasFinished = (owner, identifier) => {
+  return gdjs.TweenRuntimeBehavior._tweenList[owner][identifier].hasFinished;
+};
+
+gdjs.TweenRuntimeBehavior._setRemoveObjectFromScene = (owner, identifier, removeObjectFromScene) => {
+  return gdjs.TweenRuntimeBehavior._tweenList[owner][identifier].removeObjectFromScene = removeObjectFromScene;
+};
+
+gdjs.TweenRuntimeBehavior._removeTween = (owner, identifier) => {
+  gdjs.TweenRuntimeBehavior._tweenList[owner][identifier].instance.dispose();
+  delete gdjs.TweenRuntimeBehavior._tweenList[owner][identifier];
+};
+
+/**
+ * Add an object variable tween.
+ * @param {object} owner The runtime object
+ * @param {behavior} tweenBehavior The tween behavior
+ * @param {string} identifier Unique id to idenfify the tween
+ * @param {objectvar} variable The object variable to store the tweened value
+ * @param {number} fromValue Start value
+ * @param {number} toVaue End value
+ * @param {string} easingValue Type of easing
+ * @param {number} durationValue Duration in milliseconds
+ * @param {boolean} destroyObjectWhenFinished Destroy this object when the tween ends
+ */
+gdjs.TweenRuntimeBehavior.addObjectVariableTween = function (
+  owner,
+  tweenBehavior,
+  identifier,
+  variable,
+  fromValue,
+  toValue,
+  easingValue,
+  durationValue,
+  destroyObjectWhenFinished
+) {
+
+  if (gdjs.TweenRuntimeBehavior._tweenExists(owner, identifier)) {
+    return;
+  }
+
+  let newTweenable = new shifty.Tweenable();
+
+  newTweenable.setConfig({
+    from: { value: fromValue, progress: 0.0 },
+    to: { value: toValue, progress: 1.0 },
+    duration: durationValue,
+    easing: easingValue,
+    step: state => {
+      //variable.setNumber(state.value);
+      gdjs.TweenRuntimeBehavior._getTween(owner, identifier).progress = state.progress;
+      gdjs.TweenRuntimeBehavior._getTween(owner, identifier).currentValue = state.value;
+      console.log(state);
+    }
+  });
+
+  gdjs.TweenRuntimeBehavior._addObjectVariableTween(owner, identifier, newTweenable, variable, fromValue);
```

Comment:
Here, instead of calling `_addObjectVariableTween`, which is an auxiliary function that you created to handle list of objects, do it on your behavior directly :)

```
this._tweens[identifier] = new gdjs.TweenRuntimeBehavior.ObjectVariableTweenInstance(newTweenable, false, 0, variable, currentValue);
```

Reply:
This suppose that you created an empty map at the construction of your behavior:

```js
gdjs.TweenRuntimeBehavior = function (runtimeScene, behaviorData, owner) {
  gdjs.RuntimeBehavior.call(this, runtimeScene, behaviorData, owner);

  /** @type Object.<string, gdjs.TweenRuntimeBehavior.TweenInstance > */
  this._tweens = {}; //Map containing tween identifier as key, and tween instances as values.
};
```

(The `@type` is a bit complicated, but should work according to https://stackoverflow.com/questions/19513955/how-to-document-a-dictionary-in-jsdoc)